server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -412,14 +429,23 @@ public RemoteClusterStateManifestInfo writeIncrementalMetadata(
         uploadedMetadataResults.uploadedIndicesRoutingMetadata = remoteRoutingTableService.getAllUploadedIndicesRouting(
             previousManifest,
             uploadedMetadataResults.uploadedIndicesRoutingMetadata,
-            routingTableDiff.getDeletes()
+            routingTableIncrementalDiff.getDeletes()
+        );
+
+        ClusterStateDiffManifest clusterStateDiffManifest = new ClusterStateDiffManifest(
+            clusterState,
+            previousClusterState,
+            routingTableIncrementalDiff,
+            uploadedMetadataResults.uploadedIndicesRoutingDiffMetadata != null
+                ? uploadedMetadataResults.uploadedIndicesRoutingDiffMetadata.getUploadedFilename()
+                : null
```

Comment:
Will there be a case that we have diff but the file is null?

Reply:
It can happen in case there is only new addition of index. In that case routingTableIncrementalDiff.getDiffs() would be empty, resulting in no shard-diff upload.